examples/python/change_data_feed.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (2021) The Delta Lake Project Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, expr
+from delta.tables import DeltaTable
+import shutil
+
+
+path = "/tmp/delta-change-data-feed/student"
+otherPath = "/tmp/delta-change-data-feed/student_source"
+
+# Enable SQL commands and Update/Delete/Merge for the current spark session.
+# we need to set the following configs
+spark = SparkSession.builder \
+    .appName("Change Data Feed") \
+    .master("local[*]") \
+    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension") \
+    .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog") \
+    .getOrCreate()
+
+
+def cleanup():
+    shutil.rmtree(path, ignore_errors=True)
+    shutil.rmtree(otherPath, ignore_errors=True)
+    spark.sql("DROP TABLE IF EXISTS student")
+    spark.sql("DROP TABLE IF EXISTS student_source")
+
+
+def read_cdc_by_path(starting_version):
```

Comment:
I'd be inclined to say we don't need both `read_cdc_by_path` and `read_cdc_by_table_name`. The difference seems trivial and users should be able to extrapolate between the two. I think it just complicates the example a bit, and takes away from the focus which is reading the cdc data. What do you think?

Reply:
this sgtm.